velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +365,48 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  int precision = getDecimalPrecisionScale(*fromType).first;
+  int scale = getDecimalPrecisionScale(*fromType).second;
+  // A varchar's size is estimated with unscaled value digits, dot, leading
+  // zero, and possible minus sign.
+  int32_t rowSize = precision + 1;
+  if (scale > 0) {
+    ++rowSize; // A dot.
+  }
+  if (precision == scale) {
+    ++rowSize; // Leading zero.
+  }
+
+  auto flatResult = result->asFlatVector<StringView>();
+  Buffer* buffer =
```

Comment:
In ase when rowSize <= StringView::kInlineSize we don't need to allocate the buffer, right?

Reply:
Maybe we can write to the inline address directly just like [StringView.h#L65-L72](https://github.com/facebookincubator/velox/blob/main/velox/type/StringView.h#L65-L72) if it is public, then the string buffer is not needed. Otherwise, memcpy is needed to copy from the allocated string buffer to the inline address. How do you think?

Reply:
I think we can allocate totalBytes + inlineSize, and then reuse the allocated inline buffer.

Reply:
@mbasmanova Thank you for the informative suggestion. I implemented option (1) in this PR, and will work on option (2) in a follow up PR as you suggested.

Reply:
That sounds good to me. Thanks.

FYI, there is some information about string vectors that might be helpful in https://facebookincubator.github.io/velox/programming-guide/chapter02.html

Reply:
Thank you. This guide looks very useful to me.

Reply:
@mbasmanova @jinchengchenghh Opened https://github.com/facebookincubator/velox/pull/6689 for option (2) support.